asf_heat_pump_suitability/pipeline/reweight_epc/evaluate_reweighting.py

```diff
@@ -0,0 +1,273 @@
+            )
+            reweighted_proportions = calculate_proportions(reweighted_counts)
+            error_metrics_reweighted = get_error_metrics(
+                reweighted_counts,
+                reweighted_proportions,
+                target_counts,
+                target_proportions,
+            )
+
+            average_error_reduction = get_error_reduction(
+                orig_proportions, reweighted_proportions, target_proportions
+            )
+
+            prop_reweighted_counts_list = (
+                epc_subset.group_by(feature_name)
+                .agg(pl.col("proportional_weight").sum())
+                .to_dict()
+            )
+            prop_reweighted_counts = dict(
+                zip(
+                    prop_reweighted_counts_list[feature_name],
+                    prop_reweighted_counts_list["proportional_weight"],
+                )
+            )
+            prop_reweighted_proportions = calculate_proportions(prop_reweighted_counts)
+            error_metrics_prop_reweighted = get_error_metrics(
+                prop_reweighted_counts,
+                prop_reweighted_proportions,
+                target_counts,
+                target_proportions,
+            )
+
+        else:
+            # The weights may have been all Null
+            error_metrics_reweighted = {}
+            error_metrics_prop_reweighted = {}
+            average_error_reduction = None
+    else:
+        error_metrics_orig = {}
+        error_metrics_reweighted = {}
+        error_metrics_prop_reweighted = {}
+        average_error_reduction = None
+
+    final_metrics = {}
+    for metric in main_error_metrics:
+        final_metrics[metric] = {
+            "unweight": error_metrics_orig.get(metric, None),
+            "reweight": error_metrics_reweighted.get(metric, None),
+            "reweight-prop": error_metrics_prop_reweighted.get(metric, None),
+            "error_reduction": average_error_reduction,
+        }
+    return final_metrics
+
+
+def load_epc_reweights(
+    reweighted_dir: str = "s3://asf-heat-pump-suitability/outputs/2023_Q2_EPC_enhanced_weights.parquet",
+) -> pl.DataFrame:
+    """
+    Load and slightly clean the reweighted EPC data
+
+    Args:
+                    reweighted_dir (str): The S3 location of the EPC data enhanced with weights
+
+    Returns:
+                    reweighted_epc_df (pl.DataFrame): The EPC data enhanced with weights
+    """
+
+    reweighted_epc_df = pl.read_parquet(reweighted_dir)
+
+    # Bit of cleaning to allign with target data
+    reweighted_epc_df = reweighted_epc_df.rename({"NUMBER_HABITABLE_ROOMS": "nrooms"})
+    reweighted_epc_df = reweighted_epc_df.with_columns(
+        pl.col("nrooms").map_elements(
+            lambda x: "9+" if x >= 9 else str(x), return_dtype=pl.String
+        )
+    )
+    reweighted_epc_df = reweighted_epc_df.with_columns(
+        pl.Series(
+            name="nrooms", values=reweighted_epc_df["nrooms"].replace(None, "unknown")
```

Comment:
What's the reason for this line im curious?

Reply:
I think because there was an 'unknown' category in the target dataset